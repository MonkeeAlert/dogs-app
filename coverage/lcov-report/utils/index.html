
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.73% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>126/142</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.59% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>54/67</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.47% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>19/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.23% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>120/136</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="constants.ts"><a href="constants.ts.html">constants.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="functions.ts"><a href="functions.ts.html">functions.ts</a></td>
	<td data-value="92.62" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.62" class="pct high">92.62%</td>
	<td data-value="122" class="abs high">113/122</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="63" class="abs high">54/63</td>
	<td data-value="94.73" class="pct high">94.73%</td>
	<td data-value="19" class="abs high">18/19</td>
	<td data-value="92.24" class="pct high">92.24%</td>
	<td data-value="116" class="abs high">107/116</td>
	</tr>

<tr>
	<td class="file low" data-value="tinyUrlHelper.ts"><a href="tinyUrlHelper.ts.html">tinyUrlHelper.ts</a></td>
	<td data-value="41.66" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 41%"></div><div class="cover-empty" style="width: 59%"></div></div>
	</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	</tr>

<tr>
	<td class="file empty" data-value="types.ts"><a href="types.ts.html">types.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Fri Apr 08 2022 23:23:18 GMT+0300 (Eastern European Summer Time)
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    