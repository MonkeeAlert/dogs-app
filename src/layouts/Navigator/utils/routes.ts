export type RootStackParamList = {
  CatalogTabs: {
    search?: string;
    refresh?: boolean;
  };
  Search: {
    search?: string;
  };
  User: {};
  Gallery: {
    uri: string;
    img: string;
    size: {
      w: number;
      h: number;
    };
    isConnected?: string;
    search?: string;
  };
  Login: {};
};
