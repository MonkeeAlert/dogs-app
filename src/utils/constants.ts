export const colors = {
  white: '#fffafb',
  lightGray: '#E3F1EA',
  turquoise: '#2BD9A5',
  darkTurquoise: '#2E8A6E',
  darkGray: '#2b2c28',
  black: '#131515',
};

export const text = {
  xs: 7,
  s: 14,
  m: 21,
  l: 28,
  xl: 35,
  xxl: 42,
};

export const animationConfig = {
  duration: 200,
};

export const turquoiseGradientArray = [colors.turquoise, '#2bd9a5aa'];
