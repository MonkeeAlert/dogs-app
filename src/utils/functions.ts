export const flatTree = (o: any): string[] => {
  const _o = Object.assign({}, o);
  const keys = Object.keys(_o);

  for (let k in _o) {
    if (_o[k].length > 0) {
      let a = _o[k].map((i: string) => `${k}-${i}`);

      let idx = keys.indexOf(k);

      keys.splice(idx, 1, ...a);
    }
  }

  return keys;
};

export const findBreedInList = (list: string[], breed: string) => {
  const _arr = JSON.parse(JSON.stringify(list));
  const splitted = breed.toLowerCase().split(' ');

  if (
    _arr.indexOf(splitted.join('-')) !== -1 ||
    _arr.indexOf(splitted.reverse().join('-')) !== -1
  ) {
    const normal = _arr.indexOf(splitted.join('-'));
    const reversed = _arr.indexOf(splitted.reverse().join('-'));

    return normal !== -1 ? _arr[normal] : _arr[reversed];
  } else {
    const res = _arr.filter(w => {
      return splitted.filter(i => w.match(i)).length > 0;
    });

    return res.length === 0 ? [] : res.length === 1 ? res[0] : breed;
  }
};

export const parseImage = (s: string) => {
  const fromBreed = s.slice(s.indexOf('breeds'));
  const arr = fromBreed.split('/');

  return `${arr[1]}-${arr[2]}`;
};
