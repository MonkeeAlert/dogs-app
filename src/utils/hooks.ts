import {Platform} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';

const HEADER_EXPANDED_HEIGHT = 78 + 14;

export const useComputedHeaderValue = () => {
  const insets = useSafeAreaInsets();
  const headerAddedValue = insets.top + 7 * 2;

  const height = Platform.select({
    ios: HEADER_EXPANDED_HEIGHT + headerAddedValue,
    android: HEADER_EXPANDED_HEIGHT + insets.top + headerAddedValue,
    default: HEADER_EXPANDED_HEIGHT + headerAddedValue,
  });

  return {height};
};
