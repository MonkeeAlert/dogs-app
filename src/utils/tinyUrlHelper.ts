import axios from 'axios';
import firestore from '@react-native-firebase/firestore';

const instance = axios.create();

instance.defaults.baseURL = 'https://api.tinyurl.com';
instance.defaults.headers.post['Content-Type'] = 'application/json';

export const createTinyUrl = async (url: string) => {
  const fcData = await firestore().collection('api_keys').doc('tiny_url').get();
  const {key} = fcData.data() as any;

  if (!key && typeof key !== 'string') {
    throw new Error('key was received');
  } else {
    instance.defaults.headers.common['Authorization'] = `Bearer ${key}`;

    const {data} = await instance
      .post(`${instance.defaults.baseURL}/create`, {
        url,
        domain: 'rotf.lol',
      })
      .catch(err => {
        throw new Error(err);
      });

    return data.data.tiny_url;
  }
};
