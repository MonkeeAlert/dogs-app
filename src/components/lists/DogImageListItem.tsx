import {
  Dimensions,
  Image,
  Pressable,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';
import React, {useEffect, useState} from 'react';
import {Icon} from 'react-native-elements/dist/icons/Icon';
import {useDispatch, useSelector} from 'react-redux';
import {getDogsCatalog} from '../../redux/rootSelector';
import {
  animationConfig,
  colors,
  ErrorMessages,
  notificationRef,
} from '../../utils/constants';
import Animated, {useAnimatedStyle, withTiming} from 'react-native-reanimated';
import {saveToBookmarks} from '../../redux/actions/listActions';
import {useNavigation} from '@react-navigation/native';
import {checkImageCache, isAndroid} from '../../utils/functions';
import {checkConnection} from '../../../native-modules/InternetConnectionModuleAndroid';

interface Props {
  uri: string;
  idx: number;
}

const ICON_SIZE = 35;
const SCREEN_WIDTH = Dimensions.get('screen').width;
const SCREEN_HEIGHT = Dimensions.get('screen').height;
const HEADER_HEIGHT = 67.33333587646484;

const areEqual = (prev: Readonly<Props>, next: Readonly<Props>) => {
  return prev.uri === next.uri && prev.idx === next.idx;
};

const DogImageListItem = ({uri, idx}: Props) => {
  const {bookmarks} = useSelector(getDogsCatalog);
  const dispatch = useDispatch();
  const {navigate} = useNavigation<any>();

  const [img, setImg] = useState('');

  const isBookmarked = bookmarks.indexOf(uri) !== -1;

  const bStyle = useAnimatedStyle(() => {
    const _active = {
      opacity: withTiming(1, animationConfig),
      backgroundColor: withTiming(colors.turquoise, animationConfig),
    };

    const _default = {
      opacity: withTiming(0.75, animationConfig),
      backgroundColor: withTiming('transparent', animationConfig),
    };

    return isBookmarked ? _active : _default;
  }, [isBookmarked]);

  const handleSave = () => dispatch(saveToBookmarks(uri));

  const openGallery = async () => {
    const isConnected = await checkConnection();
    const search = uri.slice(
      uri.indexOf('breeds') + 'breeds/'.length,
      uri.lastIndexOf('/'),
    );

    Image.getSize(
      img,
      (width, height) => {
        let w = SCREEN_WIDTH;
        let h = SCREEN_HEIGHT / 1.5;

        const ratio = w / h;
        const imageRatio = width / height;

        if (imageRatio > ratio) {
          h = SCREEN_WIDTH / imageRatio + HEADER_HEIGHT;
        }

        navigate('Gallery', {uri, search, img, isConnected, size: {w, h}});
      },
      err => {
        notificationRef.current?.show(ErrorMessages.Default, 'error');
      },
    );
  };

  useEffect(() => {
    let isMounted = false;

    checkImageCache(uri).then(res => {
      if (!isMounted) {
        setImg(res);
      }
    });

    return () => {
      isMounted = true;
    };
  }, []);

  return typeof uri !== 'string' || uri === '' ? null : (
    <Animated.View
      testID={'DogImageListItem_View'}
      style={[
        styles.container,
        {
          position: 'relative',
          top: idx % 2 !== 0 ? 28 : 0,
        },
      ]}>
      <TouchableOpacity
        testID={'DogImageListItem_GalleryBtn'}
        activeOpacity={0.9}
        onPress={openGallery}>
        <Pressable onPress={handleSave}>
          <Animated.View
            testID={'DogImageListItem_Bookmarks'}
            style={[styles.icon, bStyle]}>
            <Icon
              type={'ionicon'}
              name={`bookmarks`}
              color={colors.white}
              size={16}
            />
          </Animated.View>
        </Pressable>

        {img !== '' ? (
          <Image
            testID={'DogImageListItem_Image'}
            source={{uri: img}}
            style={styles.image}
          />
        ) : (
          <View testID={'DogImageListItem_EmptyView'} style={styles.image} />
        )}
      </TouchableOpacity>
    </Animated.View>
  );
};

export default React.memo(DogImageListItem, areEqual);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginVertical: 7,
    marginHorizontal: 7,
    minHeight: 70,
    maxWidth: Dimensions.get('screen').width / 2 - 7 * 2 - 7,
  },
  image: {
    width: '100%',
    height: 200,
    borderRadius: 20,
    zIndex: -1,
    backgroundColor: colors.lightGray,
  },
  icon: {
    position: 'absolute',
    top: 7,
    right: 7,
    zIndex: 1,
    elevation: 1,
    width: ICON_SIZE,
    height: ICON_SIZE,
    borderRadius: ICON_SIZE,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
