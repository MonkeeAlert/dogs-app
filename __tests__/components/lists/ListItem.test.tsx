import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {useSelector} from 'react-redux';
import {create} from 'react-test-renderer';
import DogImageListItem from '../../../src/components/lists/ListItem';
import {useNavigation} from '@react-navigation/native';
import {mockedNavigate} from '../../../jest.setup';
import {checkImageCache} from '../../../src/utils/functions';
import {shallow} from 'enzyme';

describe('ListItem', () => {
  beforeEach(() => {
    useSelector.mockReturnValue({bookmarks: []});
  });

  it('matches snapshot', () => {
    const tree = create(<DogImageListItem uri={'foo'} idx={1} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders nothing when uri is empty', () => {
    // useSelector.mockReturnValueOnce({bookmarks: []});

    const {toJSON} = render(<DogImageListItem />);
    expect(toJSON()).toBe(null);
  });

  it('navigates to gallery when pressing on a specified button', () => {
    // useSelector.mockReturnValueOnce({bookmarks: []});

    const {getByTestId} = render(<DogImageListItem uri={'foo'} idx={1} />);
    const btn = getByTestId('DogImageListItem_GalleryBtn');

    fireEvent(btn, 'onPress');
    expect(mockedNavigate).toHaveBeenCalledTimes(1);
  });

  it('renders two images with different top style', () => {
    const {toJSON} = render(
      <>
        <DogImageListItem uri={'foo'} idx={1} />
        <DogImageListItem uri={'bar'} idx={2} />
      </>,
    );

    expect(toJSON()[0].props.style.top).toBe(28);
    expect(toJSON()[1].props.style.top).toBe(0);
  });

  it('renders component with empty box', () => {
    const wrapper = shallow(<DogImageListItem uri={'foo'} idx={1} />);
    const view = wrapper.find({testID: 'DogImageListItem_EmptyView'});

    expect(view.exists()).toBe(true);
  });
});
