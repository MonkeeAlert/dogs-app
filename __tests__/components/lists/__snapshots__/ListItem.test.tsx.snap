// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches snapshot 1`] = `
<View
  animatedStyle={
    Object {
      "value": Object {},
    }
  }
  collapsable={false}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 7,
      "marginVertical": 7,
      "maxWidth": 354,
      "minHeight": 70,
      "position": "relative",
      "top": 28,
    }
  }
  testID="DogImageListItem_View"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="DogImageListItem_GalleryBtn"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        animatedStyle={
          Object {
            "value": Object {
              "backgroundColor": "transparent",
              "opacity": 0.75,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 999,
            "elevation": 1,
            "height": 35,
            "justifyContent": "center",
            "opacity": 0.75,
            "position": "absolute",
            "right": 7,
            "top": 7,
            "width": 35,
            "zIndex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#fffafb",
                      "fontSize": 16,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                ï†“
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ededed",
          "borderRadius": 20,
          "height": 200,
          "width": "100%",
          "zIndex": -1,
        }
      }
      testID="DogImageListItem_EmptyView"
    />
  </View>
</View>
`;
